### DISCUSSION TITLE
i3 don't execute .sh files

### DISCUSSION THREAD

**Discussion by laximit** [i3/i3#4723] [Q&A]
if this was already answered could you point me where so i can delete this discussion, thanks

it's been a while since i last used linux, and I'm trying to setup a i3wm + polybar rice, but the problem is, i have a launch script for polybar that contains:
`
killall -q polybar
`
`
polybar main -c $HOME/.config/polybar/config
`

if i remember well, we can run this using `exec_always --no-startup-id $HOME/.config/polybar/launch.sh`.  
but when i reload i3, nothing happens, and when i run it from the command line, it works.
i already chmod it to x and nothing, is there something i'm missing

here are the [configs](https://github.com/i3/i3/files/7697695/configs.zip) if anyone want to check

**Comment by Airblader** [comment]
> but when i reload i3

A config _reload_ never triggers `exec(_always)`. You need an i3 in-place _restart_ (`i3 restart`).

**Reply by laximit** [reply]
tried restart, reload, rebooted my pc, and nothing

i've made a test script that just opens a terminal window, and didn't run either

**Reply by Airblader** [reply]
If you provide a log file we can see whether the command was executed (though I don't doubt that). You could try redirecting stdout and stderr to a file to see if the script throws any errors. Environment issues would be quite common.
